src/Core/src/Layouts/GridLayoutManager.cs

```diff
@@ -641,20 +647,108 @@ void ResolveStars(Definition[] defs, double availableSpace, Func<Cell, bool> cel
 				}
 				else
 				{
+					// Try to use density-aware distribution if possible
+					if (TryResolveStarsWithDensityAwareness(defs, availableSpace, starCount))
+					{
+						// Density-aware distribution was successful, definitions already set
+						return;
+					}
+
+					// Fallback to original method
 					// If we have a finite space, we can divvy it up among the full star weight
 					starSize = availableSpace / starCount;
+					
+					System.Diagnostics.Debug.WriteLine($"ResolveStars fallback: availableSpace={availableSpace}, starCount={starCount}, starSize={starSize}");
+					
+					// Apply the fallback calculation to star definitions
+					foreach (var definition in defs)
+					{
+						if (definition.IsStar)
+						{
+							// Give the star row/column the appropriate portion of the space based on its weight
+							var sizeValue = starSize * definition.GridLength.Value;
+							
+							// Use density from the constructor
+							definition.Size = new DensityValue(sizeValue, _density);
+						}
+					}
 				}
+			}
 
-				foreach (var definition in defs)
+			bool TryResolveStarsWithDensityAwareness(Definition[] defs, double availableSpace, double starCount)
+			{
+				// Use the density already obtained in the constructor
+				var density = _density;
+
+				// Only use density-aware distribution when density != 1.0
+				// This ensures backward compatibility when no scaling is needed
+				if (Math.Abs(density - 1.0) < 0.001)
 				{
-					if (definition.IsStar)
+					return false; // Use original algorithm for density = 1.0
+				}
+
+				// Create portions array for star definitions
+				var starDefs = new List<Definition>();
+				var portions = new List<double>();
+
+				foreach (var def in defs)
+				{
+					if (def.IsStar)
 					{
-						// Give the star row/column the appropriate portion of the space based on its weight
-						definition.Size = starSize * definition.GridLength.Value;
+						starDefs.Add(def);
+						portions.Add(def.GridLength.Value);
 					}
 				}
+
+				if (starDefs.Count == 0)
+					return true; // No star definitions to resolve
+
+				// Use DensityValue to distribute pixels precisely
+				var totalPixels = Math.Floor(availableSpace * density);
+				var pixelAllocations = DensityValue.DistributePixels(totalPixels, density, portions.ToArray());
+
+				// Assign DensityValue directly from pixels to avoid precision loss
+				for (int i = 0; i < starDefs.Count; i++)
+				{
+					// Create DensityValue directly from pixels - this is the source of truth
+					starDefs[i].Size = DensityValue.FromPixels(pixelAllocations[i], density);
```

Comment:
I think this code was incorrectly moved inside the else statement if `if (double.IsInfinity(availableSpace))` is true we still need this for loop to run and distribute the sizes

Reply:
Please address this comment